src/pages/Home.jsx

```diff
@@ -1,20 +1,37 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
+import { fetchMovies } from "../services/api";
 import MovieCard from "../components/MovieCard/MovieCard";
-import "./Home.css";
+import styles from "./Home.module.css";
 
-const sampleMoviePoster =
-  "https://img.freepik.com/free-vector/cinema-film-production-realistic-transparent-composition-with-isolated-image-clapper-with-empty-fields-vector-illustration_1284-66163.jpg";
+const Homepage = () => {
+  const [movies, setMovies] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
 
-const initialMovies = [
-  { id: 1, title: "Movie Title 1", url: sampleMoviePoster, liked: false },
-  { id: 2, title: "Movie Title 2", url: sampleMoviePoster, liked: false },
-  { id: 3, title: "Movie Title 3", url: sampleMoviePoster, liked: false },
-  { id: 4, title: "Movie Title 4", url: sampleMoviePoster, liked: false },
-  { id: 5, title: "Movie Title 5", url: sampleMoviePoster, liked: false },
-];
+  useEffect(() => {
+    const loadMovies = async () => {
```

Comment:
try to separate data fetching into a [custom hook](https://react.dev/learn/reusing-logic-with-custom-hooks).

It will make the code more clean, reusable and separate of concern